File: Programlama Dilllerinin Prensipleri/odev2/KISIM2/src/RastgeleKisi.c
/*
*
* @author Duhan UZUN, duhan.uzun@ogr.sakarya.edu.tr - Kadir ÇELİK, kadir.celik6@ogr.sakarya.edu.tr
* @since 26.04.2020
* <p>
* Rastgele kişi üreten ve ürettiği kişilerin kimlik numaralarını ve IMEI numaralarını kontrol eden programın rastgele kişi kısmının kaynak dosyası
* </p>
*/

#include "RastgeleKisi.h"

//RastgeleKisi yapısını oluşturacak fonksiyon yazılıyor.
RastgeleKisi RastgeleKisiOlustur(int i)
{
	RastgeleKisi this;
	//RastgeleKisi için bellekte gerekli yer ayrılıyor
	this = (RastgeleKisi)malloc(sizeof(struct RASTGELEKISI));
	this->degisken = i;
	this->kisi = KisiOlustur();
	this->rastgeleSayi = RastgeleOlustur();
	
	//Kisi yapısına atamalar yapılıyor
	this->kisi->telefon = TelefonNoOlustur(this->degisken);
	this->kisi->telefon->telefonNoUret(this->kisi->telefon);
	
	this->kisi->kimlikNo = KimlikNoOlustur(this->degisken);
	this->kisi->kimlikNo->kimlikNoUret(this->kisi->kimlikNo);
	
	//Fonksiyonlara ilgili göstericiler atanıyor
	this->rastgeleIsimUret = &_rastgeleIsimUret;
	this->rastgeleYasUret = &_rastgeleYasUret;
	this->yoket = &rastgeleKisiYoket;
	
	this->rastgeleIsimUret(this,"random_isimler.txt");
	this->rastgeleYasUret(this);
	return this;
}

//Rastgele isim ve soyisim üreten fonksiyon yazılıyor
void _rastgeleIsimUret(const RastgeleKisi this,char* dosyaAdi)
{
	//Dosya okumak için Dosya nesnesi çağrılıyor
    Dosya dosya = DosyaOlustur(dosyaAdi);
	//Programın çalıştığı andaki zaman hesaplanıyor
	clock_t current_time = time(NULL);
	
	//Hesaplanan zaman bir değişkenle çarpılarak rastgelSayi'nın kullanacağı sayı değerine atanıyor
	this->rastgeleSayi->sayi = (int)(current_time * this->degisken);
	//Dosyadan rastgele bir satır seçilerek isim ve soyisime atanıyor
    int rastgeleSayi =  this->rastgeleSayi->rastgeleUret(this->rastgeleSayi, 0, dosya->sayac);
	      
    char* isim = dosya->oku(dosya)[rastgeleSayi];
   
	char* temp =  strtok(isim," ");
	char* ad;
	char* soyad;
	
	ad=temp;
    temp = strtok(NULL, " ");
	soyad=temp;
	soyad[strlen(soyad)-1] = '\0';
    this->kisi->isim=ad;
    this->kisi->soyisim = soyad;
	dosya->yoket(dosya);
}

//Rastgele yaş üreten fonksiyon yazılıyor 
void _rastgeleYasUret(const RastgeleKisi this)
{
    this->kisi->yas = 0;
	//Programın çalıştığı andaki zaman hesaplanıyor
	clock_t current_time = time(NULL);
	
	//Hesaplanan zaman bir değişkenle çarpılarak rastgelSayi'nın kullanacağı sayı değerine atanıyor
	this->rastgeleSayi->sayi = (int)(current_time * this->degisken);
    this->kisi->yas = this->rastgeleSayi->rastgeleUret(this->rastgeleSayi, 0, 99);        
}

//RastgeleKisi yapısını yok eden metod yazılıyor
void rastgeleKisiYoket(RastgeleKisi this)
{	
	if(this == NULL) return;
	this->kisi->yoket(this->kisi);
	this->rastgeleSayi->yoket(this->rastgeleSayi);
	free(this);
}

File: Programlama Dilllerinin Prensipleri/odev2/KISIM1/RASTGELEKISIURET/src/RastgeleKisiUret/RastgeleKisi.java
/**
*
* @author Duhan UZUN, duhan.uzun@ogr.sakarya.ogr.edu.tr - Kadir ÇELİK, kadir.celik6@ogr.sakarya.edu.tr
* @since 20.04.2020
* <p>
* Rastgele kişi üreten ve ürettiği kişilerin kimlik numaralarını ve IMEI numaralarını kontrol eden kütüphanenin rastgele kişi üreten kısmı
* </p>
*/

package RastgeleKisiUret;

public class RastgeleKisi extends Kisi
{    
    private String rastgele_ad = new String();
    private String rastgele_soyisim = new String();
    private int rastgele_yas;
    
    public KimlikNo oTcNo = new KimlikNo();
    public Telefon oTelefonNo = new Telefon();

    //kurucuda gerekli atamaları yapıyoruz.
    public RastgeleKisi(String dosyaAdi)
    {
        oTcNo.kimlikNoUret();
        oTelefonNo.telefonNoUret();  
        rastgeleIsimUret(dosyaAdi);      
        super.setKimlikNo(oTcNo);
        super.setTelefonNo(oTelefonNo);
        super.setIsim(this.rastgele_ad);
        super.setSoyisim(this.rastgele_soyisim);
        super.setYas(rastgeleYasUret());
    }
    
    public void rastgeleIsimUret(String dosyaAdi)
    {
        //dosyadan gerekli verileri çekiyoruz
        DosyaOkuma dosya = new DosyaOkuma(dosyaAdi);
        Rastgele random = new Rastgele(dosya.getSayac());      
        int rastgeleSayi =  random.rastgeleUret();
        
        //dosyadan gerekli verileri çekiyoruz
        String isim = dosya.VerileriAl()[rastgeleSayi];
        
        //veriyi isim ve soyisim diye iki parçaya ayırıyoruz, fazla boşluklar varsa da sıkıntı olmayacak şekilde ayarladık.
        String isim2[] =  isim.split("\\s+");
        
        rastgele_ad=isim2[0];
        rastgele_soyisim = isim2[1];
    }
    
    public int rastgeleYasUret()
    {
        rastgele_yas = 0;
        
        Rastgele rastgeleYas = new Rastgele(100);
        rastgele_yas = rastgeleYas.rastgeleUret();
        
        return rastgele_yas;
    }
}